src/main.rs

```diff
@@ -12,147 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use buffrs::command;
-use buffrs::manifest::Manifest;
-use buffrs::package::{PackageName, PackageStore};
-use buffrs::registry::RegistryUri;
-use buffrs::{manifest::MANIFEST_FILE, package::PackageType};
-use clap::{Parser, Subcommand};
-use miette::{miette, WrapErr};
-use semver::Version;
+use std::env;
 
-#[derive(Parser)]
-#[command(author, version, about, long_about)]
-#[command(propagate_version = true)]
-struct Cli {
```

Comment:
Please keep the clap related code out of the lib.rs if possible; you can if you find this way of doing it unpleasant use binaries with modules (src/bin/cli/*) but in my personal opinion I'd vouch for as little outsourcing as possible at this point 

But inlining it into the library code is mixing the CLI layer with the actual impl code

Reply:
@mara-schulke : So my issue is that I have a "build.rs" that wants to call `buffrs`, but without downloading its binary. Moving the CLI logic into the library here allows this to be implemented easily, by simply calling the `run()` function.

I can address your concern in three ways, let me know which one is acceptable:

1. Keep the current solution, and accept that the CLI is part of the lib
2. Split buffrs into a workspace, with a core lib crate, a cli lib crate, and the binary crate. 
3. Move the CLI back into the bin crate, and force third-party build.rs to always go through executing `buffrs` as a binary,

Reply:
Just read your final comment below. I see. I'll go with the 3rd option for now until we move to bazel ourselves. Or what build system would you recommend ?

Reply:
I can wholeheartedly recommend you to look into nix for build orchestration; buffrs already has a nix integration so you can download and build all your protos with buffrs using nix today;

You can absolutely go with the 3rd option for your use case but my above point is that that is also not a desirable end state (none of the three are) but it may be a good compromise for backwards compat.

So how we do it is (using nix)

1. Download protos using the buffrs nix lib
2. Run build of project using cargos nix lib

In your build.rs you would then only call out to `tonic_build` as the protos are already present in the file system.

Both step 1 and 2 can be executed with a single nix build command!

Reply:
68d5f43ea1c48bc79dbf42eea3eca62dfcfd055e